src/features/web-compat.js

```diff
@@ -512,15 +512,41 @@ export default class WebCompat extends ContentFeature {
     }
 
     viewportWidthFix () {
-        const viewportTag = document.querySelector('meta[name=viewport]')
-        if (!viewportTag) return
-        const viewportContent = viewportTag.getAttribute('content')
-        if (!viewportContent) return
-        const viewportContentParts = viewportContent.split(',')
-        const widthPart = viewportContentParts.find((part) => part.includes('width'))
-        // If we already have a width, don't add one
-        if (widthPart) return
-        viewportTag.setAttribute('content', `${viewportContent},width=device-width`)
+        const viewportTags = document.querySelectorAll('meta[name=viewport]')
+        // Chrome respects only the last viewport tag
+        let viewportTag = viewportTags.length === 0 ? null : viewportTags[viewportTags.length - 1]
+        const viewportContent = viewportTag?.getAttribute('content')
+        const viewportContentParts = viewportContent?.split(/,|;/)
+        if (!viewportTag || this.desktopModeEnabled) {
+            // force wide viewport width
+            if (!viewportTag) {
+                viewportTag = document.createElement('meta')
+                viewportTag.setAttribute('name', 'viewport')
+            }
+            const forcedWidth = screen.width >= 1280 ? 1280 : 980
+            // Race condition: depending on the loading state of the page, initial scale may or may not be respected, so the page may look zoomed-in after applying this hack.
+            // Usually this is just an annoyance, but it may be a bigger issue if user-scalable=no is set, so we remove it too.
```

Comment:
To reproduce this: go to reddit.com, enable desktop mode, and sometimes the page would be zoomed in.
I tried to delay based on `document.loadingState`, but that didn't work unfortunately. Suggestions welcome.

Reply:
Without digging into the specific case I don't know if I can think of anything.

For Android I don't think we can run any earlier even if we moved the code into `load` phase rather than `init`.